docs/how-to/end-to-end-testing.mdx

```diff
@@ -0,0 +1,56 @@
+# End to End Testing
+
+Part of the software development lifecycle for many organizations includes the need to do end to end (e2e) testing.
+This is often done to test for regressions but can also be used to validate that acceptance criteria has been met.
+
+# Testing Basic Applications
+The Zephyr build output provides a link to the immutable version that was created and published as part of the build.
+If you have not set up an environment that will automatically get this version you can proceed testing this URL itself.
+
+We found one of the quickest and easiest ways to do this is to get the output URL.
+You can leverage the build output directly and using regex find the appropriate output text.
+
+```yml title=pull_request.yml
+run: |
+  OUTPUT=$(npm run build)
+  echo "$OUTPUT"
+  URL=$(echo "$OUTPUT" | grep -oE 'https://[^ ]+')
+  HOST=$(echo "$OUTPUT" | grep -oP 'ZEPHYR\s+\K\S+(?=#)')
+  echo "Extracted Host Zephyr URL: $URL"
+  echo "Extracted Host: $HOST"
+  echo "host=$HOST" >> "$GITHUB_OUTPUT"
+  echo "url=$URL" >> "$GITHUB_OUTPUT"
+```
+The output can then be used as variables in later steps of your pipeline to serve as the base URL for your test.
+
+```yml title=pull_request.yml
```

Comment:
This ain't pretty.
We probably could come up with some NPX script to read from `.zephyr` cache with app_uid input.

Reply:
for sure, but until we do this "works". There is a task to make it better that can be addressed.

Reply:
Got it.
Would it be good then to have something saying that this workaround is temporary or somethin along these lines?

If not, all good, resolve this comment and merge.